Assets/Scripts/Movement.cs

```diff
@@ -30,49 +32,64 @@ public class Movement : MonoBehaviour
     public Transform playerBody;
     private float X_Rotation = 0;
 
-    //For Torch Tossing\\
-    public GameObject Torch;
+    // For Torch Tossing \\
     private InputAction TossInput;
+    public GameObject Torch;
+    public GameObject TorchSpawnPoint;
+    public float UpwardsForce = 5.5f, ForwardsForce = 8f;
+    private float DefaultForwardsForce;
+    public float TorchForceMultiplier = 2f;
+    public GameObject HeldTorch;
+    private bool isTorchButtonHeldDown = false;
+    private Vector3 DefaultTorchPosition;
+
+    //  For The WHIP \\
+    private bool canFireWhip = true;
+    public float whipCoolDownTime = 1f;
+    private InputAction WhipInput;
+    public float whipRange = 5f;
+    private Vector3 hitPosition;
+    public bool didHit = false;
+    private RaycastHit hit;
 
 
     void Start()
     {
-        MoveController = playerControls.actions["Move"];
-        Look = playerControls.actions["Look"];
-        TossInput = playerControls.actions["Toss"];
-
         Cursor.lockState = CursorLockMode.Locked;
+        DefaultForwardsForce = ForwardsForce;
+        DefaultTorchPosition = HeldTorch.transform.localPosition;
     }
 
     // Update is called once per frame
     void Update()
     {
+        //Movement setup
         MoveDirection = MoveController.ReadValue<Vector2>();
         MousePosition = Look.ReadValue<Vector2>();
         isGrounded = Physics.CheckSphere(GroundCheck.position, GroundDistance, GroundMask);
 
         // Looking stuff \\
-        float Look_X = MousePosition.x * X_AimSensitivity;
-        float Look_Y = MousePosition.y * Y_AimSensitivity;
+        float Look_X = MousePosition.x * X_AimSensitivity *Time.deltaTime * 100;
+        float Look_Y = MousePosition.y * Y_AimSensitivity *Time.deltaTime * 100;
```

Comment:
Should we just multiply `X_AimSensitivity` (and `Y`) by 100 instead of doing it here?